model/project_ref.go

```diff
@@ -2434,6 +2434,44 @@ func GetMessageForPatch(patchID string) (string, error) {
 	return project.CommitQueue.Message, nil
 }
 
+func ValidateContainers(project string, containers []Container) error {
+	pRef, err := FindMergedProjectRef(project, "", true)
```

Comment:
I don't completely understand what this merging project ref logic does when passed a version, so this comment might be wrong.

It seems like `ValidateContainers` might need a specific version's config rather than assuming that it's not associated with a version. For example, [generate tasks](https://github.com/evergreen-ci/evergreen/blob/c2a10070a720bc20018b7a0bed8b40b44820834e/units/generate_tasks.go#L162-L172), passes in the project and its particular version's project ref. `CheckProjectErrors` does not take the project ref into account (although `CheckProjectSettings` respects the passed-in project ref). Will this still validate correctly for a generated task that might add more configuration dynamically?

If this relies on the project ref to do validation, it seems like `validateContainers` could be part of `projectSettingsValidators` instead of `projectErrorValidators` so that the correct project ref gets passed into this validation without needing to do a lookup and missing the version-specific config.

Reply:
Moving into `projectSettingsValidators ` sgtm so we can remove the need for this DB call to retrieve the project ref.  The function call has been removed but to explain, the empty string is used for merging with a version controlled project config, where an empty string will cause the ref to be merged with the most recent good version of the project config. It doesn't signify an empty version. In some cases such as specific tasks / revisions a version is passed in, but in this case I was just merging with the most recent project config.